### PR TITLE
[Arm64] Implement Simd.Insert

### PR DESCRIPTION
**Pull request by sdmaclea**
@tannergooding @CarolEidt PTAL

### CODE REVIEW

**Comment by sdmaclea** [comment]
This is passing all tests except it hits an ABI issue when returning a Vector64<T> when the immediate index is out of range.

I will address that in a separate pull request.

**Comment by sdmaclea** [comment]
My test case forgot to check the contained Extract case.  It is failing due to a LSRA issue.  I will amend the patch.

**Comment by tannergooding** [comment]
@sdmaclea, I did a bit of work to get HWIntrinsic nodes to support being marked as contained here: https://github.com/dotnet/coreclr/pull/16095

Some of it is applicable to ARM/ARM64 as well.

**Comment by sdmaclea** [comment]
@tannergooding Thanks.  Looks like I need a few lines from #16095.  I seems like temporarily disabling the contain code would be prudent.

**Comment by sdmaclea** [comment]
I disabled containment.  All tests are passing with the ABI changes which I will push separately.

This should pass regression since previous version also passed.

@CarolEidt PTAL

**Comment by sdmaclea** [comment]
Pushed a change to re-enable containment now that it is working.

File: src/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -901,6 +909,26 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
...
 
+        case HWIntrinsicInfo::SimdInsertOp:
+            if (op2->IsCnsIntOrI())
+            {
+                MakeSrcContained(node, op2);
+
+                GenTree* op3 = argList->Rest()->Rest()->Current();
+
+                // TODO-ARM64-CQ Support containing NI_ARM64_SIMD_GetItem (vector element to element move)
+                if (op3->OperIs(GT_HWIntrinsic) && (op3->AsHWIntrinsic()->gtHWIntrinsicId == NI_ARM64_SIMD_GetItem))
```
Comment (CarolEidt) [inline src/jit/lowerarmarch.cpp]: These needs a comment explaining what is being done here (i.e. folding an extract of an element with an insert of that element into another vector).

File: src/jit/codegenarm64.cpp [hunk 2]
```diff
@@ -5301,9 +5301,110 @@ void CodeGen::genHWIntrinsicSimdExtractOp(GenTreeHWIntrinsic* node)
...
+    if (targetReg != op1Reg)
+    {
+        bool     is16Byte = (node->gtSIMDSize > 8);
+        emitAttr attr     = is16Byte ? EA_16BYTE : EA_8BYTE;
+        getEmitter()->emitIns_R_R(INS_mov, baseTypeSize, targetReg, op1Reg);
+    }
+
+    if (op3->isContained())
+    {
+        // Handle vector element to vector element case
```
Comment (CarolEidt) [inline src/jit/codegenarm64.cpp]: This needs a more explanatory comment to describe at a high level what's being done (e.g. this is a folding of `SimdInsertOp(GetItem)`, and to clarify why 1) `op3` is known (asserted) to be a `GetItem`, and 2) its `srcLane` is also known to be a constant.

**Comment by sdmaclea** [comment]
@CarolEidt Added comments per you request

**Pull request merged**